File: invesdwin-scripting-parent/invesdwin-scripting-julia/invesdwin-scripting-julia-runtime-contract/src/test/java/de/invesdwin/scripting/julia/runtime/contract/callback/ParametersAndReturnsTestShort.jl
println("getShort")
if isdefined(Main, :getShort) && !isnothing(getShort)
	error("getShort already defined!")
end
getShort = callback("getShort")
println(typeof(getShort))
println(getShort)
if typeof(getShort) != Int16
	error("getShort not Int16!")
end
callback("setShort",getShort)

println("getShortVector")
if isdefined(Main, :getShortVector) && !isnothing(getShortVector)
	error("getShortVector already defined!")
end
getShortVector = callback("getShortVector")
println(eltype(getShortVector))
println(getShortVector)
if eltype(getShortVector) != Int16
	error("getShortVector not Int16!")
end
callback("setShortVector",getShortVector)

println("getShortVectorAsList")
if isdefined(Main, :getShortVectorAsList) && !isnothing(getShortVectorAsList)
	error("getShortVectorAsList already defined!")
end
getShortVectorAsList = callback("getShortVectorAsList")
println(eltype(getShortVectorAsList))
println(getShortVectorAsList)
if eltype(getShortVectorAsList) != Int16
	error("getShortVectorAsList not Int16!")
end
callback("setShortVectorAsList",getShortVectorAsList)

println("getShortMatrix")
if isdefined(Main, :getShortMatrix) && !isnothing(getShortMatrix)
	error("getShortMatrix already defined!")
end
getShortMatrix = callback("getShortMatrix")
println(eltype(getShortMatrix))
println(getShortMatrix)
if eltype(getShortMatrix) != Int16
	error("getShortMatrix not Int16!")
end
callback("setShortMatrix",getShortMatrix)

println("getShortMatrixAsList")
if isdefined(Main, :getShortMatrixAsList) && !isnothing(getShortMatrixAsList)
	error("getShortMatrixAsList already defined!")
end
getShortMatrixAsList = callback("getShortMatrixAsList")
println(eltype(getShortMatrixAsList))
println(getShortMatrixAsList)
if eltype(getShortMatrixAsList) != Int16
	error("getShortMatrixAsList not Int16!")
end
callback("setShortMatrixAsList",getShortMatrixAsList)

File: invesdwin-scripting-parent/invesdwin-scripting-runtime-lua/src/test/java/de/invesdwin/scripting/runtime/lua/tests/callback/ParametersAndReturnsTestShort.lua
arraysToString = java.method(java.import('de.invesdwin.util.collections.Arrays'),'toString','java.lang.Object')

print("getShort")
if(getShort ~= nil) then
	error("getShort already defined!")
end
getShort = callback("getShort")
getShortType = type(getShort)
print(getShortType)
print(getShort)
if(getShortType ~= "number") then
	error("getShort not Short!")
end
callback("setShort", getShort)

print("getShortVector")
if(getShortVector ~= nil) then
	error("getShortVector already defined!")
end
getShortVector = callback("getShortVector")
getShortVectorType = type(getShortVector[1])
print(getShortVectorType)
print(arraysToString(getShortVector))
if(getShortVectorType ~= "number") then
	error("getShortVector not Short!")
end
callback("setShortVector", getShortVector)

print("getShortVectorAsList")
if(getShortVectorAsList ~= nil) then
	error("getShortVectorAsList already defined!")
end
getShortVectorAsList = callback("getShortVectorAsList")
getShortVectorAsListType = type(getShortVectorAsList[1])
print(getShortVectorAsListType)
print(arraysToString(getShortVectorAsList))
if(getShortVectorAsListType ~= "number") then
	error("getShortVectorAsList not Short!")
end
callback("setShortVectorAsList", getShortVectorAsList)

print("getShortMatrix")
if(getShortMatrix ~= nil) then
	error("getShortMatrix already defined!")
end
getShortMatrix = callback("getShortMatrix")
getShortMatrixType = type(getShortMatrix[1][1])
print(getShortMatrixType)
print(arraysToString(getShortMatrix))
if(getShortMatrixType ~= "number") then
	error("getShortMatrix not Short!")
end
callback("setShortMatrix", getShortMatrix)

print("getShortMatrixAsList")
if(getShortMatrixAsList ~= nil) then
	error("getShortMatrixAsList already defined!")
end
getShortMatrixAsList = callback("getShortMatrixAsList")
getShortMatrixAsListType = type(getShortMatrixAsList[1][1])
print(getShortMatrixAsListType)
print(arraysToString(getShortMatrixAsList))
if(getShortMatrixAsListType ~= "number") then
	error("getShortMatrixAsList not Short!")
end
callback("setShortMatrixAsList", getShortMatrixAsList)
